rispy/writer.py

```diff
@@ -16,26 +17,61 @@ def invert_dictionary(mapping):
     return remap
 
 
+class WriterImplementation(str, Enum):
```

Comment:
same comments in the parser file would apply here. Remove `WriterImplemenation, make mappings required instead of optional in BaseWriter, move mapping property to init, etc